src/Pagination/index.jsx

```diff
@@ -483,40 +116,34 @@ Pagination.propTypes = {
    * string, symbol, etc. Default is chevrons rendered using fa-css.
    */
   icons: PropTypes.shape({
-    leftIcon: PropTypes.node,
-    rightIcon: PropTypes.node,
+    leftIcon: PropTypes.oneOfType([PropTypes.element, PropTypes.func]),
+    rightIcon: PropTypes.oneOfType([PropTypes.element, PropTypes.func]),
```

Comment:
[clarification] Is this intended to be a breaking change from what's supported today? In the current state, consumers can technically pass any node rather than just an element. 

That said, it only looks like this impacts 1 usage of `Pagination` though as a breaking change according to usage insights; I think it's probably fine to go this route. That 1 usage should be using the `buttonLabels` instead of passing a node with `<Icon>` and `<span className="sr-only">...</span>` elements. Everything in the `sr-only` span really should probably be in the `buttonLabels` prop.

Reply:
Good catch! That was a while ago, took me some time to remember why I did this haha, I was definitely trying to avoid making breaking changes, but here it's almost unavoidable I think.

The reason is that in original implementation these icon props worked only for the `default` variant of `Pagination`, i.e. they were ignored for all other variants and you could not override icons for `secondary` variant for example. I felt like this definitely should be fixed during this refactoring and I don't really see any other way, because we use `IconButton`s for other variants and they expect our icon instances, not any node.

We could probably add some magic with our `withDeprecatedProps` HOC to support both behaviours, but I think that would be pretty ugly... not sure if it's worth doing that, I feel like this breaking change is pretty minor and we can go with it to keep this component clean.